flatland/core/transition_map.py

```diff
@@ -266,9 +266,38 @@ def set_transition(self, cell_id, transition_index, new_transition, remove_deade
         send_infrastructure_data_change_signal_to_reset_lru_cache()
         #assert len(cell_id) == 3, \
         #    'GridTransitionMap.set_transition() ERROR: cell_id tuple must have length 3.'
+        
+        nDir = cell_id[2]
+        #print(cell_id, type(nDir))
+        if type(nDir) == np.ndarray:
+            # I can't work out how to dump a complete backtrace here
+            #import traceback
+            #try:
+            #    assert type(nDir)==int, "cell direction is not an int"
+            #except Exception as e:
+            #    traceback.print_exception(e)
+            #    traceback.print_tb(e.__traceback__)
+            #    print(traceback.format_exc())
+            print("fixing nDir:", cell_id, nDir)
+            nDir = int(nDir[0])
+
+        if type(transition_index) is not int:
+            print("fixing transition_index:", cell_id, transition_index)
+            if type(transition_index) == np.ndarray:
+                transition_index = int(transition_index[0])
+            else:
+                print("transition_index type:", type(transition_index))
+                transition_index = int(transition_index)
+            
+        if type(new_transition) is not int:
+            print("fixing new_transition:", cell_id, new_transition)
+            new_transition = int(new_transition[0])
+
+        #print("fixed:", cell_id, type(nDir), transition_index, new_transition, remove_deadends)
+
```

Comment:
No.  That's why I was trying to make it generate a stack backtrace - so I could see where it was being called from.  Hopefully I'm just missing something obvious.
It may well be some of my own logic passing on numpy single-element arrays; or perhaps an upstream change which some of my logic passes through.  I think Adrian spent some time removing the numpy types for performance. 
Numpy has introduced a change which generates a deprecation warning about the implicit conversion, and this messes up a warning capture in one of the tests. The test generates a warning deliberately and then tries to catch it.   Perhaps we could make the warning capture a bit more robust.  But we still need to remove this implicit conversion.